src/TypeScriptGenerator.ts

```diff
@@ -249,12 +249,16 @@ function selectionsToAST(
     // It might be some other type then the listed concrete types. We try to
     // figure out which types remain here.
     let possibleTypesLeft: TypeID[] | null = null;
-    const innerType = nodeType !== null ? schema.getRawType(nodeType) : null;
-    if (innerType !== null && schema.isUnion(innerType)) {
+    const innerType =
+      nodeType !== null ? schema.getNullableType(nodeType) : null;
+    if (
+      innerType !== null &&
+      (schema.isUnion(innerType) || schema.isInterface(innerType))
```

Comment:
I’m not sure if I got it. What unions have in common with interfaces?

Reply:
Both can resolve to a set of other concrete types. So in `fragment X on MyType { __typename }` both in the case of a union and an interface the value of `__typename` can be a set of values. For the union it’s the types of the union, and for the interface it’s the types that implement the interface.